src/modifier/default.rs

```diff
@@ -0,0 +1,537 @@
+        (e, Expression::Atom(Atom::Variable(a)), Expression::Atom(Atom::Variable(b))) => (e.clone() * Expression::Atom(Atom::Variable(*a.min(b))) * Expression::Atom(Atom::Variable(*a.max(b))), *a.min(b) != *a),
+        (e1, e2, e3) => (e1.clone() * e2.clone() * e3.clone(), false),
+    }
+}
+
+fn denegate_internal(map: &LinearMap<Atom, Expression, 8>) -> (Expression, bool) {
+    match map.get(&Atom::Escape('A', 1)).unwrap() {
+        Expression::Atom(Atom::Numeric(n)) => match n {
+            Numeric::Integer(i) => if i < &0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Integer(-i))))), true)
+            } else {
+                (Expression::Atom(Atom::Numeric(Numeric::Integer(*i))), false)
+            },
+            Numeric::Decimal(f) => if f < &0.0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Decimal(-f))))), true)
+            } else {
+                (Expression::Atom(Atom::Numeric(Numeric::Decimal(*f))), false)
+            },
+            Numeric::Fraction(r1, r2) => if r1 < &0 && r2 < &0 {
+                (Expression::Atom(Atom::Numeric(Numeric::Fraction(-r1, -r2))), true)
+            } else if r1 < &0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Fraction(-r1, *r2))))), true)
+            } else if r2 < &0 {
+                (Expression::Negate(Box::new(Expression::Atom(Atom::Numeric(Numeric::Fraction(*r1, -r2))))), true)
+            } else {
+                (Expression::Atom(Atom::Numeric(Numeric::Fraction(*r1, *r2))), false)
+            },
+        },
+        e => (e.clone(), false),
+    }
+}
+
+//an AdaptableModifier that reduces identities in an expression tree
+pub fn reduce() -> AdaptableModifier {
+    AdaptableModifier::from_str_list(vec![
+        ("_*1 + _*1", "2 * _*1"),
+        ("_A1 * _*1 + _*1", "(1 + _A1) * _*1"),
+        ("_*1 + _A1 * _*1", "(1 + _A1) * _*1"),
+        ("_A1 * _*1 + _A2 * _*1", "(_A1 + _A2) * _*1"),
+
+        ("_*1 + _*2 + _*2", "_*1 + 2 * _*2"),
+        ("_*1 + _A1 * _*2 + _*2", "_*1 + (1 + _A1) * _*2"),
+        ("_*1 + _*2 + _A1 * _*2", "_*1 + (1 + _A1) * _*2"),
+        ("_*1 + _A1 * _*2 + _A2 * _*2", "_*1 + (_A1 + _A2) * _*2"),
+
+        ("_*1 - _*1", "0"),
+        ("_A1 * _*1 - _*1", "(_A1 - 1) * _*1"),
+        ("_*1 - _A1 * _*1", "(-_A1 + 1) * _*1"),
+        ("_A1 * _*1 - _A2 * _*1", "(_A1 - _A2) * _*1"),
+
+        ("_*1 + _*2 - _*2", "_*1"),
+        ("_*1 + _A1 * _*2 - _*2", "_*1 + (_A1 - 1) * _*2"),
+        ("_*1 + _*2 - _A1 * _*2", "_*1 + (-_A1 + 1) * _*2"),
+        ("_*1 + _A1 * _*2 - _A2 * _*2", "_*1 + (_A1 - _A2) * _*2"),
+
+        ("--_*1", "_*1"),
+
+        ("0 + _*1", "_*1"),
+        ("_*1 + -_*2", "_*1 - _*2"),
+        ("-_*1 + _*2", "_*2 - _*1"),
+
+        ("1 * _*1", "_*1"),
+        ("0 * _*1", "0"),
+        ("-_*1 * _*2", "-(_*1 * _*2)"),
```

Comment:
Shouldn't we also have `-a * -b` -> `a * b`?

Reply:
yes, I'll add that

Reply:
actually, it just needs the inverse, not the double. if we have (a*-b)->-(a*b), (-a*b)->-(a*b), and --a->a, it's covered implicitly.

Reply:
that seems like a lot of unnecessary passes of the modifier though.

Reply:
that's true